tests/avocado/test_ovmf_with_240_vcpus.py

```diff
@@ -0,0 +1,67 @@
+import avocado
+import os
+import shutil
+import sys
+
+from base import VM
+
+
+class TestOvmfVcpus(avocado.Test):
+    """
+    Run with:
+
+        avocado run test_ovmf_with_240_vcpus.py \
+        -m test_ovmf_with_240_vcpus.py.data/parameters.yaml
+
+    :avocado: tags=ovmf
+    """
+
+    def setUp(self):
+        ovmf_code_path = self.params.get('OVMF_CODE')
+        ovmf_vars_path = self.params.get('OVMF_VARS')
+        if not ovmf_code_path or not os.path.exists(ovmf_code_path):
+            basename = os.path.basename(__file__)
+            self.cancel('OVMF_CODE file not found. Set the correct '
+                        'path on "%s.data/parameters.yaml" and run this test '
+                        'with: "avocado run %s -m %s.data/parameters.yaml"' %
+                        (basename, basename, basename))
+        if not ovmf_vars_path or not os.path.exists(ovmf_vars_path):
+            basename = os.path.basename(__file__)
+            self.cancel('OVMF_VARS file not found. Set the correct '
+                        'path on "%s.data/parameters.yaml" and run this test '
+                        'with: "avocado run %s -m %s.data/parameters.yaml"' %
+                        (basename, basename, basename))
+
+        ovmf_code_tmp = os.path.join(self.teststmpdir,
+                                     os.path.basename(ovmf_code_path))
+        ovmf_vars_tmp = os.path.join(self.teststmpdir,
+                                     os.path.basename(ovmf_vars_path))
+        if not os.path.exists(ovmf_code_tmp):
+            shutil.copy(ovmf_code_path, self.teststmpdir)
+        if not os.path.exists(ovmf_vars_tmp):
+            shutil.copy(ovmf_vars_path, self.teststmpdir)
+
+        args = ['-smp', '240']
+        args.extend(['-drive', 'file=%s' % ovmf_code_tmp])
+        args.extend(['-drive', 'file=%s' % ovmf_vars_tmp])
+        self.vm = VM(args)
+
+    def test_run_vm(self):
+        """
+        According to the RHBZ1447027, the issue is: Guest cannot boot
+        with 240 or above vcpus when using ovmf.
+        Fixed in commit e85c0d14014514a2f0faeae5b4c23fab5b234de4.
+
+        :avocado: tags=RHBZ1447027
+        """
+
+        try:
+            self.vm.launch()
+        except:
+            self.fail("Error starting VM")
+
+    def tearDown(self):
+        self.vm.shutdown()
+
+if __name__ == "__main__":
+    avocado.main()
```

Comment:
Calling `avocado.main()` will fail because of the missing parameters... Maybe we could load \<TestName\>/parameters.yaml or \<TestName\>.yaml by default they exist.

Anyway, just a small usability tweak for us to think about. I think we need to fine-tune the multiplexer behavior if we intend to use it as the main parameter handling system. The more we use it, the more we'll understand the shortcomings.